woocommerce/woocommerce-ios
# [Order form] Add product and price properties to `CollapsibleProductRowCardViewModel`

Closes: #11423
Part of #11357
⚠️ Depends on #11416 ⚠️ 


## Why
This adds properties that are used in `CollapsibleProductRowCard` from `ProductRowViewModel` to `CollapsibleProductRowCardViewModel`. This is another step toward disentangling the logic used in `CollapsibleProductCard` from `ProductRowViewModel`, eventually allowing the products section in the order form to be displayed using `OrderItem` data.

`CollapsibleProductRowCardViewModel` is still relying on two properties from `ProductRowViewModel`:

* `discount`
* `id`

These are both being used with `ProductDiscountView`, to add/remove discounts on order items. This can be updated once `ProductDiscountView` has its own view model (in #11425).

## How

Properties that are now part of `CollapsibleProductRowCardViewModel`:

* `imageURL`
* `name`
* `skuLabel`
* `price`

These are also still properties in `ProductRowViewModel`, and for now they are still using the same data as before. However, this change will make it possible to use different data in the future (e.g. displaying the price from `OrderItem` instead of the product or variation).

This also adds `pricedIndividually` as a parameter when initializing the view model. It is still a property on `ProductRowViewModel`, which is needed until `ProductDiscountView` has its own view model (in #11425).

## Testing instructions

There should be no changes to the order form UX:

- Go to the Orders tab
- Tap on `+` to create an order
- Add at least one product to the order --> the product image, name, SKU, and price should be displayed as before
- Expand any product card --> the price row should show the product price as expected
- Tap to increment the product quantity --> the quantity and total product price in the price row should also be updated
- Tap "Add discount" --> the price row (under the product name) should show the updated quantity and price

Additional testing:

- [x] @rachelmcr tests that the price summary view still displays bundled item prices as expected.
- [x] @rachelmcr tests products with/without SKU.
- [x] @rachelmcr tests that the expected name is used for variations.


---
